x/gov/types/deposit.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Deposit
+type Deposit struct {
+	Depositor  sdk.AccAddress `json:"depositor"`   //  Address of the depositor
+	ProposalID uint64         `json:"proposal_id"` //  proposalID of the proposal
+	Amount     sdk.Coins      `json:"amount"`      //  Deposit amount
+}
+
+func (d Deposit) String() string {
+	return fmt.Sprintf("deposit by %s on Proposal %d is for the amount %s",
+		d.Depositor, d.ProposalID, d.Amount)
+}
+
+// Deposits is a collection of Deposit objects
+type Deposits []Deposit
+
+func (d Deposits) String() string {
+	if len(d) == 0 {
+		return "[]"
```

Comment:
Given that the normal `out` is very descriptive, probably also this empty array should be something like: `No deposits for Proposal %d`

Reply:
Yeah keep in mind, a lot of the "additions" in this PR are from simply moving existing logic from one file/package to another -- most of it is primarily existing code. So I don't want to litter this PR with code hygiene cleanup as it'll be just too large at that point 😄